### PR TITLE
[Docs] Readme: Contributor Instructions and Make Target

### LINKED ISSUE #220: [Docs] Contributor build, run, test instructions for MeshSync
Document the build, run and test instructions for MeshSync.

Include the enhancement to use this standard Makefile includes that we have in each of the other repos for an improved make command DevEx? 

_Originally posted by @leecalcote in https://github.com/meshery/meshsync/issues/216#issuecomment-1681731491_

---
**Contributor Resources**
- [Meshery documentation site](https://docs.meshery.io/)
- [Meshery documentation source](https://github.com/meshery/meshery/tree/master/docs)
- [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)

### PR DESCRIPTION
**Pull request by KiptoonKipkurui**
**Description**
Instructions for running meshsync locally
This PR fixes #220 
This PR has additional documentation to enable one run meshsync locally. It also updates the makefile with helper functions to aid in running meshsync locally 
**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
@KiptoonKipkurui, gooood. Perhaps, we can refactor to suggest that the Meshery Operator and Meshery Broker be deployed into a Kubernetes environment as the preferred alternative to installing locally. What do you think?

**Comment by KiptoonKipkurui** [comment]
@leecalcote am still learning the interconnection and working of Meshery Operator, The Meshery Broker and Meshsync to be able to give an informed answer at the moment, I will circle back to this when am more comfortable with the architecture

File: Makefile [hunk 1]
```diff
@@ -56,3 +56,8 @@ run: check
 .PHONY: error
 error:
 	go run github.com/layer5io/meshkit/cmd/errorutil -d . analyze -i ./helpers -o ./helpers
+
+ # runs a local instance of nats server in detached mode
+PHONY: run-nats
+run-nats:
```
Comment (leecalcote) [inline Makefile:62]:
```suggestion
PHONY: nats
nats:
```

We have a target naming convention of noun-verb (e.g. `nats-run`). In the even t of __run__, though, we simply drop it and assume __run__ is the default behavior.

**Comment by leecalcote** [comment]
> @leecalcote am still learning the interconnection and working of Meshery Operator, The Meshery Broker and Meshsync to be able to give an informed answer at the moment, I will circle back to this when am more comfortable with the architecture

Sounds real good, @KiptoonKipkurui

**Comment by Ghat0tkach** [comment]
thanks for you contribtution , @KiptoonKipkurui 

Let's discuss it on the websites call.
Please add this as an agenda item in the meeting minutes, if you would :)
https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit#heading=h.lohhtewfwima

**Review by leecalcote — APPROVED** [review]

**Comment by leecalcote** [comment]
Thank you, @KiptoonKipkurui 👍

**Pull request merged**